cli/src/command.rs

```diff
@@ -151,7 +151,7 @@ pub fn run() -> sc_cli::Result<()> {
 			runner.run_node_until_exit(|config| async move {
 				match config.role {
 					Role::Light => service::build_light(config),
-					_ => service::build_full(config, false).map(|full| full.2),
```

Comment:
check if `--dev` is passed. if not, print a warning instant sealing can only be enabled for dev mode and disable instant sealing.